nrf_rpc/nrf_rpc_cbor.c

```diff
@@ -190,43 +152,32 @@ void nrf_rpc_cbor_rsp_no_err(struct nrf_rpc_cbor_ctx *ctx)
 
 }
 
-void nrf_rpc_cbor_decoding_done(CborValue *value)
+void nrf_rpc_cbor_decoding_done(struct nrf_rpc_cbor_ctx *ctx)
 {
-	struct handler_proxy_ctx *ctx =
-		NRF_RPC_CONTAINER_OF(value, struct handler_proxy_ctx, value);
-
-	nrf_rpc_decoding_done(ctx->in_packet);
+	nrf_rpc_decoding_done(ctx->packet);
 }
 
 void _nrf_rpc_cbor_proxy_handler(const uint8_t *packet, size_t len,
 				void *handler_data)
 {
-	struct handler_proxy_ctx ctx;
+	struct nrf_rpc_cbor_ctx ctx;
 
 	ctx.in_packet = packet;
 
 	struct _nrf_rpc_cbor_decoder *cbor_handler =
 		(struct _nrf_rpc_cbor_decoder *)handler_data;
 
-	cbor_buf_reader_init(&ctx.reader, packet, len);
-
-	if (cbor_parser_init(&ctx.reader.r, 0, &ctx.parser, &ctx.value) !=
-	    CborNoError) {
+	zcbor_new_decode_state(ctx.zs, ARRAY_SIZE(ctx.zs), ctx.packet, len, 1);
 
-		if (cbor_handler->decoding_done_required) {
-			nrf_rpc_decoding_done(packet);
-		}
-		nrf_rpc_err(-NRF_EBADMSG, NRF_RPC_ERR_SRC_RECV, NULL,
-			    NRF_RPC_ID_UNKNOWN, NRF_RPC_PACKET_TYPE_CMD);
-		return;
+	if (cbor_handler->decoding_done_required) {
```

Comment:
This should be done only in case on an error. It seems like `zcbor_new_decode_state` cannot fail so probably this is not needed here. What do you think @doki-nordic ?

Reply:
Removed